### ISSUE TITLE
Document trial's coverage output format

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4512]
|<img alt="Screwtape's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Screwtape reported|
|-|-|
|Trac ID|trac#4512|
|Type|enhancement|
|Created|2010-06-20 13:36:17Z|

In #twisted this evening, somebody wanted to know where the authoritative documentation for Trial's coverage output format was. So far as I can see, it isn't documented anywhere, but it seems a legitimate addition to the `trial(1)` man page.

The attached patch includes the following changes:

 * Adds a description of the coverage output format to the section describing the `--coverage` command-line option.
 * Various things that used apostrophes front-and-rear as single-quotes have been changed to use a grave as the opening quote. *roff replaces apostrophe characters with U+2019 RIGHT SINGLE QUOTATION MARK and grave characters with U+2018 LEFT SINGLE QUOTATION MARK, so it should be slightly more pleasant to read.
 * Different parts of the man page used different conventions to describe option parameters; I think nearly every combination of bold, italic and angle-brackets was used. These have all been standardised to undecorated italic, as described in the "conventions" section of `man(1)` on my Ubuntu Lucid machine.
 * Likewise, the SYNOPSIS section was also updated to use the conventions of `man(1)`.

To check the rendering of the page, my advice (on Linux) is:
```
man -Tps ./trial.1 > /tmp/trial.ps
evince /tmp/trial.ps
```

I guess somebody should also check whether the patched man page can be read by Lore, but I'm not sure how to check that.

Attachments:

* [document-coverage-format.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/38e/38e2ffe5668ce2950089528c1ac1325595a293cf/18a61bc96a931f6e48b13128157b2ae5bdb08c86.patch) (5555 bytes) - added by Screwtape on 2010-06-20 13:36:43Z - Add details of trial's coverage format to trial(1)
* [document-coverage-format-4512.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/38e/38e2ffe5668ce2950089528c1ac1325595a293cf/2ad7247f7313cb819f2ae54689bf5ec1f6cfb5e1.patch) (6050 bytes) - added by Screwtape on 2010-06-26 14:05:53Z - Third time's the charm.

<details><summary>Searchable metadata</summary>

```
trac-id__4512 4512
type__enhancement enhancement
reporter__Screwtape Screwtape
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1277040977000000 1277040977000000
changetime__1277564023000000 1277564023000000
version__None None
owner__ 
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to 3k-|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_4"></a>|@jesstess set owner to Screwtape|
|-|-|

Thanks for working on this, Screwtape. Some review feedback:

 * "Generate  coverage information in _trial_temp/coverage/". Since --temp-directory lets you change the temp directory, can you generalize the statement about where coverage/ ends up?
 * "namewith" ==> "name with"
 * "named `twisted.trial.runner.cover'" ==> "named `twisted.trial.runner.cover'."
 * since Twisted requires Python 2.4 or higher, do we need the "Requires Python 2.3" note?
 * linewrap the sys.settrace() line
 * http://twistedmatrix.com/bugs/ should be something like http://twistedmatrix.com/trac/newticket
 * the copyright needs a bump

**Comment by twisted-trac** [comment]
|<img alt="Screwtape's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|Screwtape removed owner|
|-|-|

I've attached a patch against today's SVN trunk that includes my previous changes as well as addressing the issues found by jesstess.

A comment about the "reporting bugs" link - the standard groff "manual page" template has special markup for hyperlinks which I've used (renders as "text <url>" in for plain-text output, and as a proper link for HTML output with "man --html"). However, this isn't part of the traditional `man` markup, so I'm not sure if Lore's convert-man-to-HTML system will work with it (and I don't know how to trigger such a thing manually).

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_6"></a>|@jesstess set owner to Screwtape|
|-|-|

As it turns out, lore isn't happy with several of the changes -- sorry for not checking this with an earlier review.

Googling for how to lore man pages led me to http://twistedmatrix.com/trac/ticket/3423. Stealing from twisted.python._release, you can do something like:

```
from twisted.python.filepath import FilePath
dir = FilePath('doc/core/man')
from twisted.python._release import ManBuilder
m = ManBuilder()
m.build(dir)
```

to get troff ==> xhtml. I'll add documentation for this to the ReviewingDocumentation page.

**Comment by twisted-trac** [comment]
|<img alt="Screwtape's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_7"></a>|Screwtape removed owner|
|-|-|

The attached patch against current SVN trunk includes all my previous changes, plus:

 * After being converted to Lore format, converts cleanly to HTML with no warnings.
 * Removed the clever hyperlink thing; just state the bug-reporting URL in plain text for Lore compatibility.
 * In the `--reactor` write-up, made sure the dashes were ordinary hyphens, not em-dashes.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_9"></a>|@jesstess set status to `closed`|
|-|-|

Apply document-coverage-format-4512.patch by Screwtape.

Author: Screwtape
Reviewer: jesstess
Fixes: [#4512](https://github.com/twisted/twisted/issues/4512)

trial's coverage output format is now documented in the trial man page.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
